### ISSUE TITLE
Inclusion of the faster dot product algorithm

### ISSUE THREAD

**Issue by zacmon** [fredrik-johansson/python-flint#7]
Hi,

If I have Arb 2.18 installed, will this module incorporate the use of the fast dot product algorithm available in the latest Arb?

Thanks,
Zach

**Comment by fredrik-johansson** [comment]
Yes. There is no function at the Python level that exposes the dot product as such (e.g. for computing the dot product of two Python lists of arbs/acbs), but all Arb matrix and polynomial functions will benefit.

**Comment by zacmon** [comment]
Perhaps I should open a new issue for my follow-up question which is semi-unrelated, but I was wondering the following: would I benefit from moving a project programmed previously using the mpmath module to the python-flint module? I can't find any speed comparisons between the two, though I have read in your blog and elsewhere that you haven't developed mpmath recently, which leads me to believe I should adopt this wrapper instead. The project I have isn't necessarily complicated, but it does have many multiplications, additions, and powers. I began translating the python code to C with Arb before I realized things got very long very quickly. (To give some background, I'm making a calculator for a function which uses the higher-order chain rule for which I'm coding a look-up table for the partial bell polynomials. I've already coded it in python. Using Arb, things become very pedantic with arb_mul, arb_add, arb_power, etc. so I'd like to utilize the native speed of Arb without losing the easiness of coding in python if possible.)
